src/scales/scale.time.js

```diff
@@ -7,6 +7,59 @@ moment = typeof moment === 'function' ? moment : window.moment;
 var defaults = require('../core/core.defaults');
 var helpers = require('../helpers/index');
 
+// Integer constants are from the ES6 spec.
+var MIN_INTEGER = Number.MIN_SAFE_INTEGER || -9007199254740991;
+var MAX_INTEGER = Number.MAX_SAFE_INTEGER || 9007199254740991;
+
+var INTERVALS = {
+	millisecond: {
+		major: true,
+		size: 1,
+		steps: [1, 2, 5, 10, 20, 50, 100, 250, 500]
+	},
+	second: {
+		major: true,
+		size: 1000,
+		steps: [1, 2, 5, 10, 30]
+	},
+	minute: {
+		major: true,
+		size: 60000,
+		steps: [1, 2, 5, 10, 30]
+	},
+	hour: {
+		major: true,
+		size: 3600000,
+		steps: [1, 2, 3, 6, 12]
+	},
+	day: {
+		major: true,
+		size: 86400000,
+		steps: [1, 2, 5]
+	},
+	week: {
+		major: false,
+		size: 604800000,
+		steps: [1, 2, 3, 4]
+	},
+	month: {
+		major: true,
+		size: 2.628e9,
+		steps: [1, 2, 3]
+	},
+	quarter: {
+		major: false,
+		size: 7.884e9,
+		steps: [1, 2, 3, 4]
```

Comment:
No, 4 quarters isn't right. If you look at all the others you notice that they max out at half the possible. E.g. there are 1000ms in a second, so the max step size is 500. Similarly, we should max `quarters` out at `2`

I would also suggest adding `15` to `minute` and `second` as it'd be very natural to plot every 15 minutes (or seconds). In fact this is a much more natural way of referring to time than the current values. If someone asks you what time it is, you'll commonly say quarter past or quarter 'til.

Days is very tricky. Firstly, `week` and `quarter` are skipped for purposes of determining `majorUnit`. This is because they're uncommonly used `units`. If we showed "week 1", "week 2", etc. on the axis that'd be very strange. Instead we jump to the far more common unit `month` and show "Jan", "Feb", etc. Now the really tricky thing about this is that there's an inconsistent number of days in a month. So let's say you want to divide a month into three parts. For Feb you might plot the dates Feb 1, 10, 20 (adding 9.3 and rounding to the nearest) and for March you might plot March 1, 11, 21. This inconsistency causes other bugs like the one reported in https://github.com/chartjs/Chart.js/issues/4523. I would suggest, let's not worry about it in this PR. I plan on sending a PR with a better logic for the tick generation in order to fix https://github.com/chartjs/Chart.js/issues/4523